scripts/grading/grade_submission.py

```
@@ -0,0 +1,1635 @@
+Step 1 — Identify checklist items:
+  The rubric contains ☐ checkbox items under each criterion.
+  Locate every ☐ item for the criterion being graded.
+  If no ☐ items exist, use your holistic judgment to estimate checklist_pct.
+
+Step 2 — Evaluate each ☐ item against the student submission:
+  YES     = clearly demonstrated with evidence
+  PARTIAL = partially addressed or implied
+  NO      = missing, incorrect, or not mentioned
+
+Step 3 — Compute checklist_pct:
+  checklist_pct = (yes_count + 0.5 × partial_count) / total_items × 100
```

Ruff (0.15.9)

[warning] String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)